src/main/java/fi/aalto/cs/apluscourses/ui/utils/Bindable.java

```diff
@@ -26,8 +26,8 @@ public Bindable(@NotNull T target, @NotNull BiConsumer<T, S> targetSetter) {
    */
   public Bindable(@NotNull T target, @NotNull BiConsumer<T, S> targetSetter, boolean lateInvoke) {
     this.target = target;
-    this.targetSetter = lateInvoke
-            ? (t, s) -> SwingUtilities.invokeLater(() -> targetSetter.accept(t, s)) : targetSetter;
+    this.targetSetter = lateInvoke ? (t, s) -> ApplicationManager.getApplication()
+        .invokeLater(() -> targetSetter.accept(t, s)) : targetSetter;
```

Comment:
Just interested: did `SwingUtilities::invokeLater` not work?

Reply:
I think ApplicationManager is what intellij wants us to use

Reply:
the SwingUtilities one doesn't work all the time